src/System.Private.ServiceModel/tests/Scenarios/Extensibility/WebSockets/WebSocketTests.cs

```diff
@@ -0,0 +1,234 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using System.ServiceModel;
+using System.ServiceModel.Channels;
+using Xunit;
+
+public static class WebSocketTests
+{
+    [Fact]
+    [OuterLoop]
+    [ActiveIssue(424)]
+    public static void WebSocketHttpRequestReplyBinaryStreamed()
+    {
+        NetHttpBinding binding = null;
+        ChannelFactory<IWSRequestReplyService> channelFactory = null;
+        IWSRequestReplyService client = null;
+        FlowControlledStream uploadStream = null;
+
+        try
+        {
+            // *** SETUP *** \\
+            binding = new NetHttpBinding();
+            binding.WebSocketSettings.TransportUsage = WebSocketTransportUsage.Always;
+            binding.MaxReceivedMessageSize = ScenarioTestHelpers.DefaultMaxReceivedMessageSize;
+            binding.MaxBufferSize = ScenarioTestHelpers.DefaultMaxReceivedMessageSize;
+            binding.TransferMode = TransferMode.Streamed;
+            binding.MessageEncoding = NetHttpMessageEncoding.Binary;
+
+            channelFactory = new ChannelFactory<IWSRequestReplyService>(binding, new EndpointAddress(Endpoints.WebSocketHttpRequestReplyBinaryStreamed_Address));
+            client = channelFactory.CreateChannel();
+
+            // *** EXECUTE *** \\
+            using (Stream stream = client.DownloadStream())
+            {
+                int readResult;
+                // Read from the stream, 1000 bytes at a time.
+                byte[] buffer = new byte[1000];
+
+                do
+                {
+                    readResult = stream.Read(buffer, 0, buffer.Length);
+                }
+                while (readResult != 0);
+            }
+
+            uploadStream = new FlowControlledStream();
+            uploadStream.ReadThrottle = TimeSpan.FromMilliseconds(500);
```

Comment:
FlowControlledStream is a Stream which is IDisposable.  So it should be put in a 'using' or disposed in the finally block